lib/reek/source/source_code.rb

```diff
@@ -37,9 +37,10 @@ def initialize(code, description, parser = Parser::Ruby22)
       # @return an instance of SourceCode
       def self.from(source)
         case source
-        when File   then new(source.read, source.path)
-        when IO     then new(source.readlines.join, 'STDIN')
-        when String then new(source, 'string')
+        when File     then new(source.read, source.path)
+        when IO       then new(source.readlines.join, 'STDIN')
+        when Pathname then new(source.read, source.to_s)
```

Comment:
Hm, to me `Pathname` is a representation of a path, so reading from (and writing to) it makes sense to me.

I agree the fewer options to `SourceCode.from` the better, but I also believe `Pathname` is strictly superior to `String` for representation of paths. I’m reluctant to drop `String` from here as `Examiner#initialize` calls `Source::SourceCode.from(source)` directly on whatever it gets as its first parameter, and – for better or worse ;) – we expose `Examiner.new` as part of our current API. Accepting both (for now) here seems like a lesser evil.